core/src/main/groovy/grails/views/mvc/renderer/DefaultViewRenderer.groovy

```diff
@@ -73,14 +73,21 @@ abstract class DefaultViewRenderer<T> extends DefaultHtmlRenderer<T> {
         }
 
         def webRequest = ((ServletRenderContext) context).getWebRequest()
-        if (webRequest.controllerNamespace) {
-            viewUri = "/${webRequest.controllerNamespace}" + viewUri
-        }
-
         def request = webRequest.currentRequest
         def response = webRequest.currentResponse
 
-        AbstractUrlBasedView view = (AbstractUrlBasedView)viewResolver.resolveView(viewUri, request, response)
+        AbstractUrlBasedView view
+        String namespace = webRequest.controllerNamespace
+        if (namespace) {
+            view = (AbstractUrlBasedView)viewResolver.resolveView("/${namespace}${viewUri}", request, response)
+
+            if (view == null) {
```

Comment:
this if block could be moved outside of this if to remove the duplication of lines 85 and 88